core/templates/dev/head/domain/utilities/browser-checker.service.ts

```diff
@@ -26,21 +26,19 @@ const CONSTANTS = require('constants.ts');
 })
 export class BrowserCheckerService {
   private _supportsSpeechSynthesis(): boolean {
-    var supportLang = false;
     if (window.hasOwnProperty('speechSynthesis')) {
-      // TODO(#7165): Replace any with correct type.
-      speechSynthesis.getVoices().forEach((voice: any) => {
+      speechSynthesis.getVoices().forEach((voice: SpeechSynthesisVoice) => {
         CONSTANTS.AUTOGENERATED_AUDIO_LANGUAGES.forEach(
           (audioLanguage: any) => {
             if (voice.lang === audioLanguage.speech_synthesis_code ||
               (this._isMobileDevice() &&
               voice.lang === audioLanguage.speech_synthesis_code_mobile)) {
-              supportLang = true;
+              return true;
             }
           });
       });
     }
-    return supportLang;
+    return false;
```

Comment:
Actually I'm a bit confused, reading this again. Doesn't this always return false? I don't see a case in which it returns true any more, given the changes that you have made to the function.

I.e., nothing seems to be done with the return value from the internal forEach() call.

Reply:
The intention here wasn’t to change functionality. I just noticed that we’re setting a flag in the inner most loop that never gets changed so I made it return true when the conditions inside were true.

Reply:
Ah -- actually, if you do "return true" inside a for loop, it doesn't pass anything outward. See https://stackoverflow.com/questions/34653612/what-does-return-keyword-mean-inside-foreach-function

You might want to use `.some()` instead?

Reply:
Ah, I didn't know that, thanks for bringing that up!

I went went ahead and used `.some()` instead.

Reply:
Done!